diff --git a/node_modules/@ledgerhq/hw-transport-node-hid/package.json b/node_modules/@ledgerhq/hw-transport-node-hid/package.json
index 5d89f41..507455a 100644
--- a/node_modules/@ledgerhq/hw-transport-node-hid/package.json
+++ b/node_modules/@ledgerhq/hw-transport-node-hid/package.json
@@ -28,7 +28,7 @@
   "dependencies": {
     "@ledgerhq/devices": "8.0.0-next.1",
     "@ledgerhq/hw-transport": "6.27.29",
-    "node-hid": "2.1.2",
     "semver": "7.3.5"
   },
   "optionalDependencies": {
diff --git a/node_modules/@ledgerhq/hw-transport-node-hid/src/NodeHid.js b/node_modules/@ledgerhq/hw-transport-node-hid/src/NodeHid.js
index b315c61..27b9c9e 100644
--- a/node_modules/@ledgerhq/hwCannot read properties of undefined (reading 'indexOf')/hw-transport-node-hid/src/NodeHid.js
+++ b/node_modules/@ledgerhq/hw-transport-node-hid/src/NodeHid.js
@@ -10,13 +10,12 @@
 const VID = 0x2c97;
 
 function getDevices() {
-  // $FlowFixMe
-  return HID.devices(VID, 0x0);
+  return [];
 }
 
 function getDevicePath(device) {
   if (device.product.includes("Nano S")) {
-    // $FlowFixMe
-    const [_, serial] = HID.devices(VID, 0x0).find(d => d.interface === 0 && d.usage === 65280) || [];
+    const [_, serial] = [].find(d => d.interface === 0 && d.usage === 65280) || [];
     return serial;
   }
 
